### PR TITLE
remove compaction timeout checker in datacoord

### PR DESCRIPTION
**Pull request by jiaoew1991**
Signed-off-by: Enwei Jiao <enwei.jiao@zilliz.com>
issue: #18674

### CODE REVIEW

**Comment by sunby** [comment]
@jiaoew1991 If a task failed, we should notify the datacoord to set the segment compaction flag to flase. 
@xiaofan-luan Do we need the timeout state for compaction task? It is hard to control the timeout time because of the unstable segments count of a task. We can control the compaction parallism on datanode and a task will either succeed evendually or failed due to crash. If we can control the scale of a task, we will get a resonable time spent on this task.

File: internal/datanode/data_node.go [hunk 1]
```diff
@@ -820,6 +819,39 @@ func (node *DataNode) Compaction(ctx context.Context, req *datapb.CompactionPlan
...
+			Status: &commonpb.Status{
+				ErrorCode: commonpb.ErrorCode_Success,
+			},
+			State:  commonpb.CompactionState_Completed,
+			PlanID: k.(UniqueID),
+			Result: v.(*datapb.CompactionResult),
+		})
+		return true
+	})
+	node.compactionExecutor.cleanCompleted()
```
Comment (congqixia) [inline internal/datanode/data_node.go:848]: This block may has concurrency problem. If a compaction plan completed between `iteration` and `cleanCompleted`, its result will be missed.

File: internal/datacoord/session_manager.go [hunk 2]
```diff
@@ -216,6 +220,53 @@ func (c *SessionManager) execReCollectSegmentStats(ctx context.Context, nodeID i
...
+					SourceID: Params.DataCoordCfg.GetNodeID(),
+				},
+			})
+			if err != nil {
+				log.Info("Get State failed", zap.Error(err))
+				return
+			}
+			for _, rst := range resp.GetResults() {
+				if rst.GetStatus().GetErrorCode() != commonpb.ErrorCode_Success {
+					continue
```
Comment (congqixia) [inline internal/datacoord/session_manager.go:252]: It's weird to has state code in result struct

**Comment by congqixia** [comment]
/cc @ThreadDao 
Need your opinion about the change in tests.

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by jiaoew1991** [comment]
rerun ut

**Comment by jiaoew1991** [comment]
rerun ut

File: internal/datanode/data_node.go [hunk 3]
```diff
@@ -820,6 +819,33 @@ func (node *DataNode) Compaction(ctx context.Context, req *datapb.CompactionPlan
 	}, nil
 }
 
+// GetCompactionState called by DataCoord
+// return status of all compaction plans
+func (node *DataNode) GetCompactionState(ctx context.Context, req *datapb.CompactionStateRequest) (*datapb.CompactionStateResponse, error) {
+	log.Info("DataNode.GetCompactionState")
+	results := make([]*datapb.CompactionStateResult, 0)
+	node.compactionExecutor.executing.Range(func(k, v any) bool {
```
Comment (congqixia) [inline internal/datanode/data_node.go:827]: Maybe add datanode healthy check before processing the compaction data?

File: internal/datacoord/compaction.go [hunk 4]
```diff
@@ -266,6 +305,24 @@ func (c *compactionPlanHandler) isTimeout(now Timestamp, start Timestamp, timeou
...
+	c.parallelCh[nodeID] <- struct{}{}
+}
+
+func (c *compactionPlanHandler) releaseQueue(nodeID int64) {
+	log.Debug("try to release queue", zap.Int64("nodeID", nodeID))
+	ch, ok := c.parallelCh[nodeID]
+	if !ok {
+		return
+	}
+	<-ch
```
Comment (congqixia) [inline internal/datacoord/compaction.go:324]: concurrent Read/Write on map may panic

File: internal/datacoord/compaction.go [hunk 5]
```diff
@@ -219,6 +242,8 @@ func (c *compactionPlanHandler) completeCompaction(result *datapb.CompactionResu
 	}
 	// TODO: when to clean task list
 
+	nodeID := c.plans[planID].dataNodeID
+	c.releaseQueue(nodeID)
 	return nil
```
Comment (congqixia) [inline internal/datacoord/compaction.go:247]: Maybe the mutex is still needed here?

Comment (jiaoew1991) [reply internal/datacoord/compaction.go:247]: It's called from mutex wrapper, cannot add mutex again

**Comment by jiaoew1991** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**